src/test/java/service/impl/FileReaderImplTest.java

```diff
@@ -0,0 +1,49 @@
+package service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import service.FileReader;
+
+class FileReaderImplTest {
+    private FileReader fileReader;
+    private Path filePath;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        fileReader = new FileReaderImpl();
+        filePath = Files.createTempFile("reportToRead", ".csv");
+        Files.write(filePath, List.of("type,fruit,quantity", "b,banana,20", "b,apple,100"));
```

Comment:
do you really need to create it before each? Also, place all test files in the test/resources directory